[DOC][MRG] Add glossary (#2731)
nilearn/nilearn: doc/glossary.rst

Glossary
========

.. currentmodule:: nilearn

The Glossary provides short definitions of neuro-imaging concepts as well
as Nilearn specific vocabulary. 

If you wish to add a missing term, please `create a new issue`_ or
`open a Pull Request`_.

.. glossary::
    :sorted:

    
    ANOVA
    Anova

>         Analysis of variance.

Reviewer: wikipedia page ?

    AUC

>         Area under the curve.

Reviewer: Link to sklearn's documentation ?

    BIDS
    bids
        `Brain Imaging Data Structure`_ is a simple and easy to adopt way
        of organizing neuroimaging and behavioral data.
    
    BOLD
    Bold
    bold

>         Blood oxygenation level dependent.

Reviewer: ```suggestion
        Blood oxygenation level dependent. This is the kind of signal measured by functional Magnetic Resonance Imaging.
```

    CanICA

>         Canonical independant component analysis.

Reviewer: Link to the paper that introduced it ?

    contrast
    contrasts
        A `contrast`_ is a linear combination of variables (parameters or
        statistics) whose coefficients add up to zero, allowing comparison
        of different treatments.

    Decoding
        `Decoding`_ consists in predicting, from brain images, the conditions
        associated to trial.

    EEG

>         Electroencephalography.

Reviewer: Lin to wikipedia paage + add one descritpive sentence ?

    
    EPI

>         Echo-Planar Imaging.

Reviewer: ```suggestion
        Echo-Planar Imaging. This is the type of sequence used to acquire functional or diffusion MRI data.
```

    FDR correction
        `False discovery rate`_ controlling procedures are designed to control
        the expected proportion of "discoveries" (rejected null hypotheses)
        that are false (incorrect rejections of the null).

    FIR

>         Finite impulse response.

Reviewer: This is a type of free-form temporal filter that is used to link neural activity with hemodynamic response, when there is uncertainty on the true model.

    fMRI
    fmri
        Functional magnetic resonance imaging is based on the fact that 
        when local neural activity increases, increases in metabolism and
        blood flow lead to fluctuations of the relative concentrations of
        oxyhaemoglobin (the red cells in the blood that carry oxygen) and
        deoxyhaemoglobin (the same red cells after they have delivered the
        oxygen). Oxyhaemoglobin and deoxyhaemoglobin have different magnetic
        properties (diamagnetic and paramagnetic, respectively), and they
        affect the local magnetic field in different ways.
        The signal picked up by the MRI scanner is sensitive to these
        modifications of the local magnetic field.

    FPR correction
    FPR
    fpr

>         False positive rate correction.

Reviewer: Needs a definition

    FREM
        `FREM`_ means "Fast ensembling of REgularized Models". It uses an implicit
        spatial regularization through fast clustering and aggregates a high 
        number of estimators trained on various splits of the training set, thus 
        returning a very robust decoder at a lower computational cost than other 
        spatially regularized methods. 

    functional connectivity
    functional connectome
        A `functional connectome`_ is a set of connections representing brain 
        interactions between regions.

    FWER correction
        `Family-wise error rate`_ is the probability of making one or more
        false discoveries, or type I errors when performing multiple
        hypotheses tests.

    GLM

>         General Linear Model.

Reviewer: This is the name of the models traditionally fit to fMRI data, where one linear model is fit to each voxel time course.

    HRF
    hrf

>         Haemodynamic response function.

Reviewer: ```suggestion
        Haemodynamic response function. This is a temporal filter that converts neural signal to hemodynamic signals observable with fMRI.
```

    ICA

>         Independant component analysis.

Reviewer: Link needed, maybe wikipedia or sklearn.

    MEG

>         Magnetoencephalography.

Reviewer: idem, wikipedia + short sentence

    MNI
        MNI stands for "Montreal Neurological Institute". Usually, this is
        used to reference the MNI space/template. The current standard MNI 
        template is the ICBM152, which is the average of 152 normal MRI scans 
        that have been matched to the MNI305 using a 9 parameter affine transform.

    MVPA

>         Mutli-Voxel Pattern Analysis.

Reviewer: ```suggestion
        Mutli-Voxel Pattern Analysis. This is the way supervised learning methods are called in the field of brain imaging.
```

    Neurovault
        `Neurovault`_ is a public repository of unthresholded statistical maps, 
        parcellations, and atlases of the human brain.

    parcellation
    parcellations
    brain parcellation
    brain parcellations
        TODO.

    predictive modelling
        TODO.

    ReNA
        TODO.

    resting-state
        TODO.

    ROC
        The `receiver operating characteristic curve`_ plots the true positive rate
        (TPR) against the false positive rate (FPR) at various threshold settings.

    searchlight
    Searchlight
        `Searchlight analysis`_ consists of scanning the brain with a searchlight. 
        That is, a ball of given radius is scanned across the brain volume and the 
        prediction accuracy of a classifier trained on the corresponding voxels is measured.

    SpaceNet
        `SpaceNet`_ is a decoder implementing spatial penalties which improve brain 
        decoding power as well as decoder maps.

    SPM

>         Statistical Parametric Mapping.

Reviewer: This is both a conceptual framework and a software. Please provide pointers.

        or target, y, from data X. Typically, we start from labeled data (the 
        training set). We need to know the y for each instance of X in order to 
        train the model. Once learned, this model is then applied to new unlabeled 
        data (the test set) to predict the labels (although we actually know them).
        There are essentially two possible types of problems:

        .. glossary::

            regression
                 In regression problems, the objective is to predict a continuous 
                 variable, such as participant age, from the data X.

            classification
                In classification problems, the objective is to predict a binary 
                variable that splits the observations into two groups, such as 
                patients versus controls.

        In neuroimaging research, supervised learning is typically used to derive an 
        underlying cognitive process (e.g., emotional versus non-emotional theory of 
        mind), a behavioral variable (e.g., reaction time or IQ), or diagnosis status 
        (e.g., schizophrenia versus healthy) from brain images.

    SVM

>         TODO.

Reviewer: link to kslearn + 1 short sentence

    TR

>         Repetition time.

Reviewer: maybe reuse BIDS definition here

        blood flow lead to fluctuations of the relative concentrations of
        oxyhaemoglobin (the red cells in the blood that carry oxygen) and
        deoxyhaemoglobin (the same red cells after they have delivered the
        oxygen). Oxyhaemoglobin and deoxyhaemoglobin have different magnetic
        properties (diamagnetic and paramagnetic, respectively), and they
        affect the local magnetic field in different ways.
        The signal picked up by the MRI scanner is sensitive to these
        modifications of the local magnetic field.

    FPR correction
        False positive rate correction. This refers to the methods employed to
        correct false positive rates such as the Bonferroni correction which
        divides the significance level by the number of comparisions made.

    FREM
        `FREM`_ means "Fast ensembling of REgularized Models". It uses an implicit
        spatial regularization through fast clustering and aggregates a high 
        number of estimators trained on various splits of the training set, thus 
        returning a very robust decoder at a lower computational cost than other 
        spatially regularized methods. 

>     functional connectivity
>     functional connectome
>         A `functional connectome`_ is a set of connections representing brain 
>         interactions between regions.

Reviewer 2: ```suggestion
    functional connectivity
        Functional connectivity is a measure of the similarity of the response patterns in two or more regions.
        A `functional connectome`_ is the set of connections representing brain interactions as measured by functional connectivity between regions.
```

    HRF
        Haemodynamic response function. This is a temporal filter that converts
        neural signals to hemodynamic signals observable with :term:`fMRI`.

    ICA
        `Independent component analysis`_ is a computational method for separating
        a multivariate signal into additive subcomponents.

    MEG
        `Magnetoencephalography`_ is a functional neuroimaging technique for mapping
        brain activity by recording magnetic fields produced by electrical currents 
        occurring naturally in the brain.

    MNI
        MNI stands for "Montreal Neurological Institute". Usually, this is
        used to reference the MNI space/template. The current standard MNI 
        template is the ICBM152, which is the average of 152 normal MRI scans 
        that have been matched to the MNI305 using a 9 parameter affine transform.

    MVPA
        Mutli-Voxel Pattern Analysis. This is the way :term:`supervised learning`
        methods are called in the field of brain imaging.

    Neurovault
        `Neurovault`_ is a public repository of unthresholded statistical maps, 
        parcellations, and atlases of the human brain.

>     parcellation
>     brain parcellation
>         Act of dividing the brain into smaller regions, i.e. parcels, in order to
>         gain a better understanding of complex human behaviors.

Reviewer 2: ```suggestion
    parcellation
        Act of dividing the brain into smaller regions, i.e. parcels. Parcellations can be defined by many
        different criteria including anatomical or functional characteristics. Parcellations can either be composed of
        "hard" deterministic parcels with no overlap between individual regions or "soft" probabilistic parcels with
        a non-zero probability of overlap. 
```

        blood flow lead to fluctuations of the relative concentrations of
        oxyhaemoglobin (the red cells in the blood that carry oxygen) and
        deoxyhaemoglobin (the same red cells after they have delivered the
        oxygen). Oxyhaemoglobin and deoxyhaemoglobin have different magnetic
        properties (diamagnetic and paramagnetic, respectively), and they
        affect the local magnetic field in different ways.
        The signal picked up by the MRI scanner is sensitive to these
        modifications of the local magnetic field.

    FPR correction
    FPR
    fpr
        False positive rate correction. This refers to the methods employed to
        correct false positive rates such as the Bonferroni correction which
        divides the significance level by the number of comparisions made.

    FREM
        `FREM`_ means "Fast ensembling of REgularized Models". It uses an implicit
        spatial regularization through fast clustering and aggregates a high 
        number of estimators trained on various splits of the training set, thus 
        returning a very robust decoder at a lower computational cost than other 

>         spatially regularized methods. 

Reviewer: I think that you should cite the paper: https://www.sciencedirect.com/science/article/abs/pii/S1053811917308182

    functional connectivity
    functional connectome
        A `functional connectome`_ is a set of connections representing brain 
        interactions between regions.

    FWER correction
        `Family-wise error rate`_ is the probability of making one or more
        false discoveries, or type I errors when performing multiple
        hypotheses tests.

    GLM
        General Linear Model. This is the name of the models traditionally fit
        to fMRI data, where one linear model is fit to each voxel time course.

    HRF
    hrf
        Haemodynamic response function. This is a temporal filter that converts
        neural signals to hemodynamic signals observable with :term:`fMRI`.

    ICA

>         `Independant component analysis`_ is a computational method for separating

Reviewer: ```suggestion
        `Independent component analysis`_ is a computational method for separating
```

        a multivariate signal into additive subcomponents.

    MEG
        `Magnetoencephalography`_ is a functional neuroimaging technique for mapping
        brain activity by recording magnetic fields produced by electrical currents 
        occurring naturally in the brain.

    MNI
        MNI stands for "Montreal Neurological Institute". Usually, this is
        used to reference the MNI space/template. The current standard MNI 
        template is the ICBM152, which is the average of 152 normal MRI scans 
        that have been matched to the MNI305 using a 9 parameter affine transform.

    MVPA
        Mutli-Voxel Pattern Analysis. This is the way :term:`supervised learning`
        methods are called in the field of brain imaging.

    Neurovault
        `Neurovault`_ is a public repository of unthresholded statistical maps, 
        parcellations, and atlases of the human brain.

    parcellation
    parcellations
    brain parcellation
    brain parcellations

>         TODO.

Reviewer: Can we take care of the remaining TODO's ? Do you need help for that ?

        then clusters that have similar signal (parameters, measurements or time courses).

.. LINKS

.. _`create a new issue`:
    https://github.com/nilearn/nilearn/issues/new/choose

.. _`open a Pull Request`:
    https://github.com/nilearn/nilearn/compare

.. _`Analysis of variance`:
    https://en.wikipedia.org/wiki/Analysis_of_variance

.. _`Area under the curve`:
    https://scikit-learn.org/stable/modules/model_evaluation.html#roc-metrics

.. _`Brain Imaging Data Structure`:
    https://bids.neuroimaging.io/

.. _`Canonical independant component analysis`:
    https://arxiv.org/abs/1006.2300

.. _`contrast`:
    https://en.wikipedia.org/wiki/Contrast_(statistics)

.. _`Decoding`:
    https://nilearn.github.io/decoding/decoding_intro.html

.. _`Electroencephalography`:
    https://en.wikipedia.org/wiki/Electroencephalography

.. _`False discovery rate`:
    https://en.wikipedia.org/wiki/False_discovery_rate

.. _`Family-wise error rate`:
    https://en.wikipedia.org/wiki/Family-wise_error_rate

.. _`FREM`:
    https://nilearn.github.io/decoding/frem.html

.. _`functional connectome`:
    https://nilearn.github.io/connectivity/functional_connectomes.html

> .. _`Independant component analysis`:

Reviewer: ```suggestion
.. _`Independent component analysis`:
```
